rs/registry/canister/src/mutations/node_management/common.rs

```diff
@@ -80,6 +80,34 @@ pub fn get_node_operator_id_for_node(
         )
 }
 
+pub fn get_node_provider_id_for_operator_id(
+    registry: &Registry,
+    node_operator_id: PrincipalId,
+) -> Result<PrincipalId, String> {
+    let node_operator_key = make_node_operator_record_key(node_operator_id);
+    registry
+        .get(node_operator_key.as_bytes(), registry.latest_version())
+        .map_or(
+            Err(format!(
+                "Node Operator Id {:} not found in the registry.",
+                node_operator_key
+            )),
+            |result| {
+                PrincipalId::try_from(
+                    NodeOperatorRecord::decode(result.value.as_slice())
+                        .unwrap()
```

Comment:
So this will panic, instead of returning an error, right?  Seems like you want to actually return the error below here?

Reply:
It's a different unwrap/message, but you're right, let's return an error for both.